agent/consul/leader_metrics.go

```diff
@@ -120,21 +124,51 @@ type certExpirationMonitor struct {
 
 const certExpirationMonitorInterval = time.Hour
 
-func (m certExpirationMonitor) monitor(ctx context.Context) error {
+func (m CertExpirationMonitor) Monitor(ctx context.Context) error {
 	ticker := time.NewTicker(certExpirationMonitorInterval)
 	defer ticker.Stop()
 
+	logger := m.Logger.With("metric", strings.Join(m.Key, "."))
+
 	for {
 		select {
 		case <-ctx.Done():
 			return nil
 		case <-ticker.C:
```

Comment:
Will this mean that the metric/logs do not get emitted until the monitoring interval has passed (so after an hour)? Or does the ticker fire immediately and then again after each interval?

Reply:
It does not fire immediately. I think that would be a good improvement to handle the scenario you mentioned.

Reply:
Added that in #10771.